### PR TITLE
Added iOS chat example

### PR DESCRIPTION
**Pull request by mrajatttt**
*Issue #, if available:*

*Description of changes:*
Added iOS chat example. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: iOSChatExample/iOSChatExample/Models/ChatManager.swift [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+    
+    
+    // MARK: - API and SDK calls
+    
+    /// Initiates a flow to start a new chat for the customer. Response of this API provides a token required to obtain credentials from the CreateParticipantConnection API in the Amazon Connect Participant Service.
+    /// - Parameters:
+    ///  - name: Information identifying the participant as part of  ParticipantDetails Object.
+    ///  - Attributes  : A custom key-value pair using an attribute map
+    ///  - ContactFlowId  : The identifier of the flow for initiating the chat. To see the ContactFlowId in the Amazon Connect console user interface, on the navigation menu go to Routing, Contact Flows.     Choose the flow. On the flow page, under the name of the flow, choose Show additional flow information. The ContactFlowId is the last part of the ARN.
+    ///  - InstanceId  : The identifier of the Amazon Connect instance. You can find the instance ID in the Amazon Resource Name (ARN) of the instance.
```
Comment (spencerlepine) [inline iOSChatExample/iOSChatExample/Models/ChatManager.swift:115]: suggestion: could we also link to [Admin guide docs](https://docs.aws.amazon.com/connect/latest/APIReference/API_StartChatContact.html) or the IOS SDK docs

**Review by spencerlepine — CHANGES_REQUESTED** [review]

File: iOSChatExample/README.md [hunk 2]
```diff
@@ -0,0 +1,115 @@
+# iOS Native Chat Demo 📱
```
[the remark this answers is not in the document]
Comment (mrajatttt) [reply iOSChatExample/README.md:1]: Yes once this branch is merged, I will raise one more PR. same goes with all other links (files, demo video)

File: iOSChatExample/README.md [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
```
[the remark this answers is not in the document]
Comment (mrajatttt) [reply iOSChatExample/README.md:2]: Same as above, once branch is merged, I will link.

File: iOSChatExample/README.md [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+ 
+An iOS example app for building custom Amazon Connect Chat. This solution implements basic [ChatWidget](https://docs.aws.amazon.com/connect/latest/adminguide/add-chat-to-website.html) functionality and is capable of Interactive Messages.
+
+> Refer to [#Specifications](#speficications) for details on compatibility, supported versions, and platforms.
+
+**Reference:**
+
+- Documentation: https://docs.aws.amazon.com/connect/latest/adminguide/enable-chat-in-app.html
+
+https://github.com/amazon-connect/amazon-connect-chat-ui-examples/assets/60903378/8887f54a-c121-4246-8981-23d05dd9fa01
```
[the remark this answers is not in the document]
Comment (mrajatttt) [reply iOSChatExample/README.md:11]: Same as above, once branch is merged, I will link.

File: iOSChatExample/Podfile.lock [hunk 5]
```diff
@@ -0,0 +1,30 @@
+PODS:
```
[the remark this answers is not in the document]
Comment (mrajatttt) [reply iOSChatExample/Podfile.lock:1]: Earlier I was hoping that i would have this example as 'install ready' but it does not make sense as it is a better practice for use to install the pods. I have updated.

File: iOSChatExample/iOSChatExample/Models/ChatManager.swift [hunk 6]
```diff
@@ -0,0 +1,203 @@
...
+    private let config = Config()
+    private let networkManager: NetworkManager
+    var websocketUrl: String?
+    var connectParticipantClient: AWSConnectParticipant?
+    var participantToken: String?
+    var connectionToken: String?
+
+    init() {
+        networkManager = NetworkManager(httpClient: DefaultHttpClient())
+        let credentials = AWSStaticCredentialsProvider(accessKey: "", secretKey: "")
```
[the remark this answers is not in the document]
Comment (mrajatttt) [reply iOSChatExample/iOSChatExample/Models/ChatManager.swift:18]: No, they do not have to.

File: iOSChatExample/iOSChatExample.xcodeproj/project.pbxproj
Comment (spencerlepine) [inline iOSChatExample/iOSChatExample.xcodeproj/project.pbxproj:1]: do we need to commit this? is it for local workspace?

**Review by spencerlepine — APPROVED** [review]

**Review by haomingli2020 — APPROVED** [review]

**Pull request merged**